packages/core/src/provider/base.ts

```diff
@@ -235,7 +264,9 @@ export class SmartAccountProvider<
       this.gasEstimator,
       this.feeDataGetter,
       this.paymasterDataMiddleware,
-      this.customMiddleware ?? noOpMiddleware
+      this.customMiddleware ?? noOpMiddleware,
+      // This applies the overrides if they've been passed in
+      async (struct) => ({ ...struct, ...overrides })
```

Comment:
note: the reason we don't short circuit the feeDataGetter if the override is passed in is because it's possible to pass in only a partial override

Reply:
why is it async? do we also need to await if so?

Reply:
It's async because the signature of the pipeAsync requires the functions to be of type `T => Promise<T>` which is what this is doing. Making it `async` implies that the response is wrapped in promise